src/python/bot/tasks/testcase_manager.py

```diff
@@ -487,37 +490,56 @@ def run_testcase_and_return_result_in_queue(crash_queue,
                    'run_testcase_and_return_result_in_queue.')
 
 
-def test_for_crash_with_retries(testcase,
-                                testcase_path,
-                                test_timeout,
-                                http_flag=False,
-                                compare_crash=True):
-  """Test for a crash and return crash parameters like crash type, crash state,
-  crash stacktrace, etc."""
-  # Cleanup any existing application instances and user profile directories.
-  # Cleaning up temp clears user profile directories and should be done before
-  # calling |get_command_line_for_application| call since that creates
-  # dependencies in the profile folder.
-  process_handler.terminate_stale_application_instances()
-  shell.clear_temp_directory()
-
-  app_directory = environment.get_value('APP_DIR')
-  command = get_command_line_for_application(
-      testcase_path, needs_http=http_flag)
-  crash_retries = environment.get_value('CRASH_RETRIES')
-  flaky_stacktrace = testcase.flaky_stack
-  warmup_timeout = environment.get_value('WARMUP_TIMEOUT')
+class TestcaseRunner(object):
+  """Testcase runner."""
+
+  def __init__(self,
+               fuzzer_name,
+               testcase_path,
+               test_timeout,
+               gestures,
+               needs_http=False):
+    self._testcase_path = testcase_path
+    self._needs_http = needs_http
+    self._test_timeout = test_timeout
+    self._gestures = gestures
+
+    engine_impl = engine.get(fuzzer_name)
+    if engine_impl:
+      self._is_black_box = False
+      self._engine_impl = engine_impl
+
+      # Read target_name + args from flags file.
+      additional_command_line_flags = get_additional_command_line_flags(
+          testcase_path)
+      self._arguments = additional_command_line_flags.split()
+      target_name = self._arguments.pop(0)
+
+      build_dir = environment.get_value('BUILD_DIR')
+      self._target_path = engine_common.find_fuzzer_path(build_dir, target_name)
+      if not self._target_path:
+        raise RuntimeError('Failed to find target ' + target_name)
```

Comment:
This is bad for regression testing, can happen in cases where target is not there in old builds, see comment at end of find_fuzzer_path. We need to find a way to bail out and not retry.

Reply:
Handled this exception in the callers.

Reply:
Thanks!